doc/specs/#13666 - VT Sequence for Screen Reader Control.md

```diff
@@ -15,58 +15,227 @@ Now that Windows Terminal has a UI Automation implementation that can be leverag
+> - `params`: none
+
+#### "Option" Role
+This is used for selectable items. The ARIA documentation can be found [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/option_role);
+> - `Ps`: `option`
+> - `params`:
+>    - `selected`: `true` if the option is currently selected. `false` otherwise. (default: `false`)
+>    - `checked`: describes the checked state when options are used in a multiple selection fashion. Supports `true`, `false`, and `mixed`. (default: `null`)
+>    - `posinset`: defines the element's number or position in the current set of options. Particularly useful if only a subset of items are visible. (default: `-1`, which represents unknown)
+>    - `setsize`: defines the number of items in the current set of options. (default: `-1`, which represents unknown)
+
+#### "Suggestion" Role
+This is used for proposed changes to an editable area. The ARIA documentation can be found [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/suggestion_role).
+> - `Ps`: `suggestion`
+> - `params`: none
+
+<!--
+>    - `role`: `insertion` if the provided text will be inserted into the text input area. `deletion` if the provided text will be removed from the text input area. (default: `insertion`)
+-->
+
+The ARIA spec has `insertion` and `deletion` as child elements for a `suggestion` tag. However, this spec has been simplified for the terminal so that terminals don't need to keep track of nested roles.
+
+#### "Cell" Role
+This is used for cells in a tabular container. The ARIA documentation can be found [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/cell_role).
+> - `Ps`: `cell`
+> - `params`:
+>    - `rowheader`: a header for the corresponding row of the current cell. (default: `""`)
+>    - `columnheader`: a header for the corresponding column of the current cell. (default: `""`)
+>    - `rowindex`: the index for the corresponding row of the current cell. (default: `null`)
+>    - `colindex`: the index for the corresponding column of the current cell. (default: `null`)
+>    - `rowsize`: the number of rows in the table. (default: `null`)
+>    - `colsize`: the number of columns in the table. (default: `null`)
+
+The ARIA spec relies on nested roles to provide header context to cells. this spec has been simplified for the terminal so that terminals don't need to keep track of nested roles.
+
+### Screen Reader Flag (Option 4)
+
+As mentioned earlier, `DSR` is already a standard method for command-line applications to query the capabilities of the attached terminal emulator. By claiming a value, the terminal can easily respond to let the command-line application know if a screen reader is attached or not. In the event the terminal emulator does not support this feature, no response is given, which is common practice.
+> `DSR` - Screen Reader
+> - command-line application query: `CSI ? 2577 n`
+> - terminal emulator response: `CSI ? 2577; Ps`
```

Comment:
It's also worth mentioning that the typical pattern for `DSR` sequences like this, is that the query number is a multiple of 5, and then you have separate numbers for each response, starting at a multiple of 10 (below the query number).

For example, the `DSR 5` query (operating status), responds with `DSR 0`, `DSR 1`, `DSR 2`, etc. The `DSR ? 15` query (printer port), responds with `DSR ? 10`, `DSR ? 11`, etc. There are exceptions to that rule (e.g. the `CPR` query, or the keyboard dialect query), but that's just because those don't really fit the pattern of a "status" response.

So unless we're expecting to extend this with lots of different response types, it would be more customary to use something like `DSR ? 2575` for the query, and then `DSR ? 2570` and `DSR ? 2571` for the two responses (_attached_ and _not attached_).

I know I previously argued that it's better to use a single number for finite resources, but that's just because the `OSC` numbers are a bit of nightmare in terms of conflicts, and there's no standard usage pattern. `DSR` is less of a risk, because I don't think there are any modern terminals using it (not counting XTerm's non-standard abuse).

Reply:
@carlos-zamora I think I may have confused things with all my `DSR` references above. So just to be clear, `DSR` is the shorthand name for the **D**evice **S**tatus **R**eport operation. `CSI n` is the actual escape sequence for that operation (where `CSI` is equal to `ESC [` in 7-bit mode).

So you can say `DSR ? 2575`, or `CSI ? 2575 n`, or possibly even `ESC [ ? 2575 n`, but you wouldn't say `DSR ? 2575 n`. And in this particular area of the documentation, `CSI ? 2575 n` is probably the most appropriate.

Reply:
Ah ok. My bad. Yeah I saw you using `DSR` above and thought applying the same notation in the spec would make it more clear. Thanks for the explanation!